internal/rbd/replicationcontrollerserver.go

```diff
@@ -158,6 +233,46 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 	}
+
+	if interval != "" {
+		intLevel := formatSchedulingInterval(string(interval))
+		imageSpec := newLevelSpec(rbdVol.Pool, rbdVol.RadosNamespace, rbdVol.RbdImageName)
+
+		ls := admin.NewLevelSpec(rbdVol.Pool, rbdVol.RadosNamespace, rbdVol.RbdImageName)
+		radosConn := rbdVol.conn.GetRadosConnection()
+
+		adminConn := admin.NewFromConn(radosConn).MirrorSnashotSchedule()
+		// list all the snapshot scheduling and check at least one image scheduling
+		// exists with specified interval.
+		ssList, err := adminConn.List(ls)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		for _, ss := range ssList {
+			if ss.Name == imageSpec {
+				for _, s := range ss.Schedule {
+					//  TODO Add support to check start time also.
+					// The start time is currently stored with different format
+					// in ceph. Comparison is not possible unless we know in
+					// which format ceph is storing it.
+					if s.Interval == intLevel {
+						return &replication.EnableVolumeReplicationResponse{}, nil
+					}
+				}
+			}
+		}
+		err = adminConn.Add(ls, interval, startTime)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		util.DebugLog(
+			ctx, "Added scheduling at interval %s, start time %s for volume %s",
```

Comment:
most parameters on different lines, but two on the 1st?

Reply:
missed to do it.